client/blocks/importer/wordpress/import-everything/pre-migration/index.tsx

```diff
@@ -152,93 +153,96 @@ export const PreMigrationScreen: React.FunctionComponent< PreMigrationProps > =
 		}
 	}, [ continueImport, sourceSiteId, startImport, requiresPluginUpdate ] );
 
-	function renderPreMigration() {
-		// Show a loading state when we are trying to fetch existing credentials
-		if ( ! hasLoaded || isRequestingCredentials ) {
-			return <LoadingEllipsis />;
+	// Set the render state based on the current component state
+	useEffect( () => {
```

Comment:
FYI: I kept the order of state checking, avoiding introducing the regression via this refactoring.